src/platforms/stm32/src/lib/gpio_driver.c

```diff
@@ -245,82 +245,90 @@ static term setup_gpio_pin(Context *ctx, term gpio_pin_tuple, term mode_term)
     term gpio_bank_atom = term_get_tuple_element(gpio_pin_tuple, 0);
     if (UNLIKELY(!term_is_atom(gpio_bank_atom))) {
         AVM_LOGE(TAG, "Bank parameter of pin tuple must be an atom! (a...h|k depending on board)");
-        return create_pair(ctx, ERROR_ATOM, INVALID_BANK_ATOM);
+        return create_pair(ctx, ERROR_ATOM, globalcontext_make_atom(ctx->global, ATOM_STR("\xC", "invalid_bank")));
```

Comment:
BTW, I am generally okay with defining the ATOM_STR's as static const chars, especially if they are going to be re-used in the code.  I have not measured the impact of doing so on the memory image, but I can't imagine it makes much of a difference.  (OTOH, any memory savings in embedded systems is generally welcome.  Certainly no reason to request a change, but just thought I'd mention it.

Reply:
That would have been the better way to do it, for sure. I just want back and forth rewriting how these atom tuples are created and returned that I didn't really consider that option during this last round of re-writes ;)

Reply:
GCC collects duplicated string constants and it merges them: when using  >= -O1 `-fmerge-constants` is used.

Reply:
I defined the ATOM_STR's as static const chars, should help with size and readability.